test/e2e/apps/rc.go

```diff
@@ -172,6 +196,27 @@ var _ = SIGDescribe("ReplicationController", func() {
 		rcStatus, err := f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Patch(context.TODO(), testRcName, types.StrategicMergePatchType, []byte(rcStatusPatchPayload), metav1.PatchOptions{}, "status")
 		framework.ExpectNoError(err, "Failed to patch ReplicationControllerStatus")
 		framework.ExpectEqual(rcStatus.Status.ReadyReplicas, int32(0), "ReplicationControllerStatus's readyReplicas does not equal 0")
+		ginkgo.By("waiting for RC to be modified")
+		eventFound = false
+		for watchEvent := range rcWatchChan {
+			if watchEvent.Type == watch.Modified {
+				eventFound = true
+				break
+			}
+		}
+		framework.ExpectEqual(eventFound, true, "failed to find RC %v event", watch.Modified)
+
+		eventFound = false
+		for watchEvent := range rcWatchChan {
+			rc, ok := watchEvent.Object.(*v1.ReplicationController)
+			framework.ExpectEqual(ok, true, "Unable to convert type of ReplicationController watch watchEvent")
+
+			if rc.Status.Replicas == testRcInitialReplicaCount {
+				eventFound = true
+				break
+			}
+		}
+		framework.ExpectEqual(eventFound, true, "RC does not have initial replica count")
```

Comment:
Why do we need this check / what is this trying to test?  

If this is trying to wait until the RC status has been updated to undo what we just patched in, it should be looking at `rc.Status.ReadyReplicas`